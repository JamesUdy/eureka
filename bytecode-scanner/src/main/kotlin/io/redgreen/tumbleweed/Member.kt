package io.redgreen.tumbleweed

sealed interface Member {
  val signature: Signature
}
