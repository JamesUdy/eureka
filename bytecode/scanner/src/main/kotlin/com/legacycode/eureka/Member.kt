package com.legacycode.eureka

sealed interface Member {
  val name: String
  val signature: Signature
  val owner: QualifiedType
}
