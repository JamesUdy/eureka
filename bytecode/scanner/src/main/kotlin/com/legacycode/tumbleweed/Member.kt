package com.legacycode.tumbleweed

sealed interface Member {
  val name: String
  val signature: Signature
  val owner: QualifiedType
}
