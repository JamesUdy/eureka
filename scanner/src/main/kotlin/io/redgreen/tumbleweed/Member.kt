package io.redgreen.tumbleweed

interface Member {
  val signature: String
}
