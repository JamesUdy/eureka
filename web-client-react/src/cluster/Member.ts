export class Member {
  constructor(public nodeId: string) {
    // empty
  }

  isMethod() {
    return this.nodeId.endsWith(')');
  }
}
