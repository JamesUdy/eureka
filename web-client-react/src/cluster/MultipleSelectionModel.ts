import { Member } from './Member';
import SelectionModel from './SelectionModel';

export class MultipleSelectionModel extends SelectionModel<Member[]> {
  static DEFAULT = new MultipleSelectionModel('', [], null, []);

  deselect(member: Member): MultipleSelectionModel {
    const selected = this.selected.filter((selectedMember) => selectedMember.nodeId !== member.nodeId);
    return this.copy(undefined, undefined, undefined, selected) as MultipleSelectionModel;
  }

  public override select(): SelectionModel<Member[]> {
    const selectedMembers = this.selected as Member[];
    let selected: Member[];
    if (this.focused) {
      selected = [...selectedMembers, this.focused];
    } else {
      selected = [...selectedMembers];
    }

    const currentFocusedIndex = this.searchResult.findIndex((member) => member === this.focused);
    let focused: Member | null;
    if (this.searchResult[currentFocusedIndex + 1]) {
      focused = this.searchResult[currentFocusedIndex + 1];
    } else {
      focused = null;
    }
    return this.copy(undefined, undefined, focused, selected);
  }

  deselectAll(): MultipleSelectionModel {
    return <MultipleSelectionModel>this.copy(undefined, undefined, null, []);
  }

  public override visibleSearchResult(): Member[] {
    return this.searchResult.filter((member) => !this.selected.map((member) => member.nodeId).includes(member.nodeId));
  }

  protected override copy(
    searchTerm: string = this.searchTerm,
    searchResult: Member[] = this.searchResult,
    focused: Member | null = this.focused,
    selected: Member[] = this.selected,
  ): SelectionModel<Member[]> {
    return new MultipleSelectionModel(searchTerm, searchResult, focused, selected);
  }
}
