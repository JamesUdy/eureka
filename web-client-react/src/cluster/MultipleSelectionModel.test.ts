import { MultipleSelectionModel } from './MultipleSelectionModel';
import { Member } from './Member';

describe('A multiple selection model', () => {
  it('should retain search results when an item is selected', () => {
    // given
    const multipleSelectionModel = new MultipleSelectionModel(
      'void',
      [new Member('void onCreate()'), new Member('void onPause()'), new Member('void onResume()')],
      new Member('void onCreate()'),
      [],
    );

    // when
    const memberSelectedModel = multipleSelectionModel.select();

    // then
    expect(memberSelectedModel).toMatchSnapshot();
    expect(memberSelectedModel.visibleSearchResult()).toEqual([
      new Member('void onPause()'),
      new Member('void onResume()'),
    ]);
  });
});
