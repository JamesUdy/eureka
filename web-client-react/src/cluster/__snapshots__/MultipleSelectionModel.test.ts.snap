// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`A multiple selection model should retain search results when an item is selected 1`] = `
MultipleSelectionModel {
  "focused": Member {
    "nodeId": "void onCreate()",
  },
  "searchResult": Array [
    Member {
      "nodeId": "void onCreate()",
    },
    Member {
      "nodeId": "void onPause()",
    },
    Member {
      "nodeId": "void onResume()",
    },
  ],
  "searchTerm": "void",
  "selected": Array [
    Member {
      "nodeId": "void onCreate()",
    },
  ],
}
`;
